modules/diskq/logqueue-disk-reliable.c

```diff
@@ -58,6 +58,14 @@ _push_to_qbacklog_tail(LogQueueDiskReliable *self, gint64 position, LogMessage *
   log_queue_memory_usage_add(&self->super.super, log_msg_get_size(msg));
 }
 
+static inline void
+_push_to_qout_tail(LogQueueDiskReliable *self, gint64 position, LogMessage *msg,
+                   const LogPathOptions *path_options)
+{
+  _push_to_memory_queue_tail(self->qout, position, msg, path_options);
```

Comment:
I think we should not use the original path_options here, because we might end up acking the message accidentally after another refactor.

Why don't we remove the path_options parameter and use a constant "no-ack" option as the non-reliable disk buffer does?
https://github.com/syslog-ng/syslog-ng/blob/master/modules/diskq/logqueue-disk-non-reliable.c#L420-L421

Reply:
That's actually a bad idea, sorry.
We store valuable information in path_options, not just the fact whether we need to ack or not. We should have kept path_options when persisting messages on the disk buffer.
Currently, this does not cause problems, but it's not a good idea in general.

Reply:
Yep, thanks for mentioning it.